### PR TITLE
dynamic selections for density_from_Universe (issue #584)

### PR DESCRIPTION
**Pull request by orbeckst**
I manually merged @Balasubra 's PR #612 that adds dynamic selections to `analysis.density.density_from_Universe()` #584
* rewrote the history to squish commits, follow StyleGuide for commit messages, fixed minor things in docs
* added a test case (just regression test)
* use `ProgressMeter` instead of bare print
* updated the default `offset` parameter for `ProgressMeter` to correspond to new 0-based `ts.frame` counting 

This PR closes PR #612 and issue #584.

### CODE REVIEW

File: package/MDAnalysis/analysis/density.py [hunk 1]
```diff
@@ -508,15 +516,22 @@ def current_coordinates():
...
-        print("Histograming %6d atoms in frame %5d/%d  [%5.1f%%]\r" % \
-              (len(coord), ts.frame, u.trajectory.n_frames, 100.0 * ts.frame / u.trajectory.n_frames),)
-        coord = current_coordinates()
+        if update_selection:
+           group = u.select_atoms(atomselection)
+           coord=group.positions
+        else:
+           coord = current_coordinates()
+
+        pm.echo(ts.frame, n_atoms=len(coord))
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/density.py]: Please make the verbosity configurable.

Comment (orbeckst) [inline package/MDAnalysis/analysis/density.py]: good point

**Comment by orbeckst** [comment]
done

**Comment by Balasubra** [comment]
I am just curious, why should we have density_from_traj and density_from_Universe separately. Is there any difference in terms of the functionality ?.
I just read your comment that you have added slicing feature to density_from_Universe(), do we need a separate function "density_from_trajectory", then ?

File: package/MDAnalysis/analysis/density.py (hunk 1 above)
Comment (kain88-de) [inline package/MDAnalysis/analysis/density.py]: Did you forget to push the verbosity change I remarked?

**Comment by kain88-de** [comment]
@Balasubra  there is no difference in functionality between the two. 'density_from_trajectory' is actually calling 'density_from_Universe'.  We do not actually need it and should delete it. @orbeckst you can do that in this PR or we do it separately  in another.

File: package/MDAnalysis/analysis/density.py (hunk 1 above)
Comment (orbeckst) [inline package/MDAnalysis/analysis/density.py]: There's the new `quiet` keyword, which can be used to mute `ProgressMeter`, it's in 5958fa92ac0d6067d2792c34c83ad77e02a3d6de.

**Comment by orbeckst** [comment]
Thanks for the comments. I removed `density_from_trajectory()`.
I am not going to fix any QC issues because they are irrelevant here,

File: package/MDAnalysis/analysis/density.py [hunk 2]
```diff
@@ -448,8 +426,20 @@ def density_from_Universe(universe, delta=1.0, atomselection='name OH2',
...
-            With *cutoff*, select '<atomsel> NOT WITHIN <cutoff> OF <soluteselection>'
-            (Special routines that are faster than the standard AROUND selection) [0]
+            With *cutoff*, select "<atomsel> NOT WITHIN <cutoff> OF <soluteselection>"
+            (Special routines that are faster than the standard ``AROUND`` selection)
+            [0]
+      update_selection
+            Should the selection of atoms be updated for every step? [``False``]
+            - ``True``: atom selection is updated for each frame, can be slow
+            - ``False``: atoms are only selected at the beginning
+      quiet
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/density.py]: When you already touch the doc string you can change it to the numpy-style format.

**Comment by kain88-de** [comment]
LGTM. There is only the comment about the doc-string style but you can merge this without adressing it.

File: package/MDAnalysis/analysis/density.py (hunk 2 above)
Comment (orbeckst) [inline package/MDAnalysis/analysis/density.py]: Sorry, don't have the time right now.

**Pull request merged**